### PR TITLE
added tests for user controller

### PR DESCRIPTION
**Pull request by JuanSaieh**
Changes:
- Added tests for user controller

trello link: https://trello.com/c/xA0dIPr0

### CODE REVIEW

File: spec/controllers/users_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,137 @@
+require 'rails_helper'
+
+RSpec.describe UsersController, type: :controller do
+
+  describe 'GET users#index' do
+
+    before do
+      create(:user)
```
Comment (Jucerweb) [inline spec/controllers/users_controller_spec.rb]: You can use a loop to create the users you want to.

File: spec/controllers/users_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    end
+
+    it 'returns succssfully' do
+      get :index
+      expect(response).to be_successful
+    end
+
+    it 'follows the routes successfully' do
+      get :index
```
Comment (Jucerweb) [inline spec/controllers/users_controller_spec.rb]: You can add this line in the before block.

File: spec/controllers/users_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+      expect(response).to render_template("index")
+    end
+
+    it 'compares that all users in the table are equal to the index instance variable' do
+      get :index
+      expect(User.all).to match(assigns(:users))
+    end
+  end
+
+  describe 'GET#new' do
```
Comment (Jucerweb) [inline spec/controllers/users_controller_spec.rb]: `describe 'GET #new' do`

File: spec/controllers/users_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,137 @@
+require 'rails_helper'
+
+RSpec.describe UsersController, type: :controller do
+
+  describe 'GET users#index' do
```
Comment (Jucerweb) [inline spec/controllers/users_controller_spec.rb]: `  describe 'GET #index' do
`

File: spec/controllers/users_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+
+    it 'compares that all users in the table are equal to the index instance variable' do
+      get :index
+      expect(User.all).to match(assigns(:users))
+    end
+  end
+
+  describe 'GET#new' do
+    it 'returns succssfully' do
+      get :new
```
Comment (Jucerweb) [inline spec/controllers/users_controller_spec.rb]: Add this line to a before

File: spec/controllers/users_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+    it 'proves that @user was instantiated correctly' do
+      get :new
+      expect(controller.instance_variable_get(:@user)).to be_a_new(User)
+    end
+  end
+
+  describe 'POST users#create' do
+
+    context 'if user is valid' do
+
```
Comment (Jucerweb) [inline spec/controllers/users_controller_spec.rb]: Delete this space

File: spec/controllers/users_controller_spec.rb [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+    end
+
+    it 'proves that @user was instantiated correctly' do
+      get :new
+      expect(controller.instance_variable_get(:@user)).to be_a_new(User)
+    end
+  end
+
+  describe 'POST users#create' do
+
```
Comment (Jucerweb) [inline spec/controllers/users_controller_spec.rb]: Delete this space

File: spec/controllers/users_controller_spec.rb [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+        expect(response).to redirect_to(root_path)
+      end
+
+      it 'redirect 302' do
+        expect(response).to have_http_status(302)
+      end
+    end
+
+    context 'if user is invalid' do
+
```
Comment (Jucerweb) [inline spec/controllers/users_controller_spec.rb]: Delete this space

File: spec/controllers/users_controller_spec.rb [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+      expect(response).to be_successful
+    end
+
+    it 'renders show template' do
+      get :show, params: { id: User.first.id }
+      expect(response).to render_template(:show)
+    end
+
+    it 'assigns @user correctly' do
+      get :show, params: { id: User.first.id }
```
Comment (Jucerweb) [inline spec/controllers/users_controller_spec.rb]: This can be added to a before

File: spec/controllers/users_controller_spec.rb [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+      expect(controller.instance_variable_get(:@user)).to eql(User.first)
+    end
+  end
+
+  describe 'DELETE #destroy' do
+
+    before do
+      create(:user)
```
Comment (Jucerweb) [inline spec/controllers/users_controller_spec.rb]: You can use the times loop or use factory bot create_list

File: spec/controllers/users_controller_spec.rb (hunk 4 above)
Comment (JuanSaieh) [reply spec/controllers/users_controller_spec.rb]: Done!

File: spec/controllers/users_controller_spec.rb (hunk 1 above)
Comment (JuanSaieh) [reply spec/controllers/users_controller_spec.rb]: Done with create_list (FactoryBot method)

File: spec/controllers/users_controller_spec.rb (hunk 2 above)
Comment (JuanSaieh) [reply spec/controllers/users_controller_spec.rb]: Done!

File: spec/controllers/users_controller_spec.rb (hunk 3 above)
Comment (JuanSaieh) [reply spec/controllers/users_controller_spec.rb]: Done!

File: spec/controllers/users_controller_spec.rb (hunk 5 above)
Comment (JuanSaieh) [reply spec/controllers/users_controller_spec.rb]: Done!

File: spec/controllers/users_controller_spec.rb (hunk 7 above)
Comment (JuanSaieh) [reply spec/controllers/users_controller_spec.rb]: Done

File: spec/controllers/users_controller_spec.rb (hunk 6 above)
Comment (JuanSaieh) [reply spec/controllers/users_controller_spec.rb]: Done

File: spec/controllers/users_controller_spec.rb (hunk 8 above)
Comment (JuanSaieh) [reply spec/controllers/users_controller_spec.rb]: Done

File: spec/controllers/users_controller_spec.rb (hunk 9 above)
Comment (JuanSaieh) [reply spec/controllers/users_controller_spec.rb]: Done!

File: spec/controllers/users_controller_spec.rb (hunk 10 above)
Comment (JuanSaieh) [reply spec/controllers/users_controller_spec.rb]: Done!

**Pull request merged**